master/internal/task/allocation.go

```diff
@@ -49,6 +49,8 @@ type (
 		killedWhileRunning bool
 		// Marks that the trial exited successfully, but we killed some daemon containers.
 		killedDaemons bool
+		// Marks that we killed some daemon containers but after a nonzero exit.
```

Comment:
hmmm these comments look odd
e.g. `killedDaemons` doesn't mean the "trial exited successfully"

and `killedDaemonsNominal` looks like a zero exit, not a "nonzero".

Reply:
also maybe rename `killedDaemonsNominal` -> `killedDaemonsGracefully` or `killedDaemonsWithoutErr`? 
(I would struggle figuring out the meaning of the word "nominal" in such context)

Reply:
nonzero is a typo - should be zero. Will change to `gracefully`